src: pomodoro: added  --repeat-previous option to pomodoro

th-duvanel opened the pull request:
`kw pomodoro --repeat-previous`allows the user to repeat the last pomodoro argument commands used by him, without having to type it all over again.

Resolves #1009

th-duvanel commented:
Just to make a checkpoint about my progress: I exchanged some emails with @davidbtadokoro about the PR. After the next week, I will focus more on the PR and the recommendations given by him.

This PR will be finished soon. Thanks for the patience!

rodrigosiqueira reviewed (changes requested):
Hi @th-duvanel ,
First of all, thanks a lot for this PR, and I am sorry for my huge delay in providing feedback.

Anyway, I believe the best way to implement this feature is to check the latest Pomodoro in the database instead of creating a temporary file. All of this information is stored in the SQLite database. I gues @davidbtadokoro can help you with that.

Also, squash the second commit into the first one.

davidbtadokoro commented:
Hi @rodrigosiqueira. Just to update you, I had a couple of offline discussions with @th-duvanel, and we formulated a good direction for him to implement this using the latest session in the DB. Thiago already has a good looking draft, and he communicated to me that he will be able to work on it soon.

th-duvanel commented:
Hello!
Sorry for the delay, things were a bit more complicated than I thought! After some conversation (and coding) with @davidbtadokoro and the suggestions from @rodrigosiqueira, we made it to this commit.
We used the database to fetch the last session, added some autocompletions for the new function, added some tests and its documentation. It's all squashed in one commit. Any change you guys would recommend, I'm open to it.
Thanks for the help and patient this far.

rodrigosiqueira commented on tests/unit/pomodoro_test.sh:
```diff
@@ -380,4 +387,54 @@ function test_show_tags()
   assert_equals_helper 'Wrong output' "$LINENO" "$expected" "$output"
 }
 
+function test_fetch_last_pomodoro_session()
```
Create one test that validates the condition that we don't have any previous session, and the user tries to use this option. Use a new test function for that.

rodrigosiqueira commented on tests/unit/pomodoro_test.sh:
```diff
@@ -380,4 +387,54 @@ function test_show_tags()
+  local values
+  local columns='("date","time","duration","tag_name","description")'
+
+  values="('2024-05-27','17:36:20','120','fake_tag','this is not a drill')"
+  sqlite3 "${KW_DATA_DIR}/kw.db" -batch "INSERT INTO 'pomodoro_report' ${columns} VALUES ${values} ;"
+  output=$(fetch_last_pomodoro_session)
+  expected=(
+    'Last pomodoro session:'
+    '- Duration: 120s'
+    '- Tag: fake_tag'
+    '- Description: this is not a drill'
+  )
+  compare_command_sequence 'Last session' "$LINENO" 'expected' "$output"
+
+  values="('2023-05-27','14:36:20','120','true_tag','this is a drill')"
+  sqlite3 "${KW_DATA_DIR}/kw.db" -batch "INSERT INTO 'pomodoro_report' ${columns} VALUES ${values} ;"
+  output=$(fetch_last_pomodoro_session)
+  expected=(
+    'Last pomodoro session:'
+    '- Duration: 120s'
+    '- Tag: fake_tag'
+    '- Description: this is not a drill'
+  )
+  compare_command_sequence 'Older session' "$LINENO" 'expected' "$output"
```
Let's add a little bit of entropy between these tests, create some new entries between the sessions, and ensure that the `--repeat` will get the new session with different parameters (time, tag, and description).

rodrigosiqueira commented on src/pomodoro.sh:
```diff
@@ -268,6 +268,46 @@ function get_tag_name()
+  options_values['DESCRIPTION']=$(cut --delimiter='|' --fields=3 <<< "$last_pomodoro_session")
+
+  say 'Last pomodoro session:'
+  say "- Duration: ${options_values['TIMER']}"
```
I don't think that showing the session in seconds is a user-friendly approach. You can improve this output by handling it on a correct scale. For example, if it has less than 60 seconds, use seconds; if it has less than 60 minutes, keep it under minutes, and so on.

rodrigosiqueira reviewed (changes requested):
Very nice improvement!

th-duvanel commented on documentation/man/features/kw-pomodoro.rst:
```diff
@@ -7,6 +7,7 @@ kw-pomodoro - Create and manage Pomodoro timers
 SYNOPSIS
 ========
 | *kw* (*p* | *pomodoro*) (-t | \--set-timer) <time>(h | m | s) [(-g | \--tag) <tag> [(-d | \--description) <desc>] [\--verbose]]
+| *kw* (*p* | *pomodoro*) (-r | \--repeat) [\--verbose]
```
I can implement this in the future. Probably it won't be hard, but more "errors" will appear, like the ones pointed out by you in the requested changes, so I prefer to wait for the next opportunity.

rodrigosiqueira replied:
Ok, but let's update this to `--repeat-previous` or similar.

th-duvanel commented on src/pomodoro.sh:
```diff
@@ -365,6 +405,13 @@ function parse_pomodoro()
         shift 2
         ;;
+      --repeat | -r)
+        fetch_last_pomodoro_session || return "$?"
+        if [[ $(ask_yN 'Would you like to repeat this session?') =~ '0' ]]; then
+          exit
+        fi
+        shift
```
Just adding an update: me and @davidbtadokoro talked about the implementation recommended by you and he clarified your description of the "problem". No worries, I will update the commit and soon as possible.

rodrigosiqueira commented on src/pomodoro.sh:
```diff
@@ -268,6 +268,46 @@ function get_tag_name()
+    options_values['ERROR']='No previous pomodoro session found'
+    return 22 # EINVAL
+  fi
+
```
Could you address this comment?

th-duvanel commented:
Hey @rodrigosiqueira, for some reason the changes I've made didn't apply to the new commits. I'm sorry for this, I made the changes you requested more than a week ago. I think we are good now, tell me what you think about the changes.

davidbtadokoro commented:
Hi @th-duvanel, and thanks for the great improvement on this PR and continuous work! There were some comments that I felt I could incorporate myself, which I communicated to you offline, so merged into unstable branch :+1:
